config.go

```diff
@@ -117,9 +119,60 @@ func makeDockerV2S2Image(oimage *ociv1.Image) (docker.Image, error) {
 	return image, nil
 }
 
+func makeDockerV2S1Image(manifest docker.V2S1Manifest) (docker.V2Image, error) {
+	// Treat the most recent (first) item in the history as a description of the image.
+	if len(manifest.History) == 0 {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from manifest")
+	}
+	dimage := docker.V2Image{}
+	err := json.Unmarshal([]byte(manifest.History[0].V1Compatibility), &dimage)
+	if err != nil {
+		return docker.V2Image{}, err
+	}
+	if dimage.DockerVersion == "" {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from history")
+	}
+	// Build a filesystem history.
+	rootFS := &docker.V2S2RootFS{
+		Type:    docker.TypeLayers,
+		DiffIDs: []digest.Digest{},
+	}
+	history := []docker.V2S2History{}
+	for i := range manifest.FSLayers {
+		d := manifest.FSLayers[i].BlobSum
+		// Prepend this layer, because the original manifest's list has the earliest layer last.
+		rootFS.DiffIDs = append([]digest.Digest{d}, rootFS.DiffIDs...)
```

Comment:
> We compute the right values when we're exporting the image, so if nobody's going to be reading the blatantly-wrong-if-compression-was-used value, then I'd lean hard in the direction of adding a comment about it here and leaving it otherwise.

Works for me. (`docker`’s pull path does enforce that the uncompressed data match the digests in `DiffIDs`, so if it ever broke, it would be very visible.)

Or perhaps create an empty array, or an array of empty strings, to make extra certain that the incorrect values are never used by anything? Not necessary by any means.